controlplane/kubeadm/internal/cluster_test.go

```diff
@@ -242,3 +243,161 @@ func (f *fakeClient) List(_ context.Context, list runtime.Object, _ ...client.Li
 	}
 	return nil
 }
+
+func TestManagementCluster_healthCheck_NoError(t *testing.T) {
+	tests := []struct {
+		name             string
+		machineList      *clusterv1.MachineList
+		check            healthCheck
+		clusterKey       types.NamespacedName
+		controlPlaneName string
+	}{
+		{
+			name: "simple",
+			machineList: &clusterv1.MachineList{
+				Items: []clusterv1.Machine{
+					controlPlaneMachine("one"),
+					controlPlaneMachine("two"),
+					controlPlaneMachine("three"),
+				},
+			},
+			check: func(ctx context.Context) (healthCheckResult, error) {
+				return healthCheckResult{
+					"one":   nil,
+					"two":   nil,
+					"three": nil,
+				}, nil
+			},
+			clusterKey:       types.NamespacedName{Namespace: "default", Name: "cluster-name"},
+			controlPlaneName: "control-plane-name",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			m := &ManagementCluster{
+				Client: &fakeClient{list: tt.machineList},
+			}
+			if err := m.healthCheck(ctx, tt.check, tt.clusterKey, tt.controlPlaneName); err != nil {
+				t.Fatal("did not expect an error?")
+			}
+		})
+	}
+}
+
+func TestManagementCluster_healthCheck_Errors(t *testing.T) {
+	tests := []struct {
+		name             string
+		machineList      *clusterv1.MachineList
+		check            healthCheck
+		clusterKey       types.NamespacedName
+		controlPlaneName string
+	}{
+		{
+			name: "machine's node was not checked for health",
+			machineList: &clusterv1.MachineList{
+				Items: []clusterv1.Machine{
+					controlPlaneMachine("one"),
+					controlPlaneMachine("two"),
+					controlPlaneMachine("three"),
+				},
+			},
+			check: func(ctx context.Context) (healthCheckResult, error) {
+				return healthCheckResult{
+					"one": nil,
+				}, nil
+			},
+		},
+		{
+			name: "health check returns an error not related to the nodes health",
+			machineList: &clusterv1.MachineList{
+				Items: []clusterv1.Machine{
+					controlPlaneMachine("one"),
+					controlPlaneMachine("two"),
+					controlPlaneMachine("three"),
+				},
+			},
+			check: func(ctx context.Context) (healthCheckResult, error) {
+				return healthCheckResult{
+					"one":   nil,
+					"two":   errors.New(""),
+					"three": errors.New(""),
+				}, errors.New("")
```

Comment:
The problem I ran into specifically was that in this case, the "two" and "three" errors were being swallowed by the healthcheck function. 

wdyt of `errors.New("two")`, `errors.New("three")`, `errors.New("meta")`, and checking that each string (`two`, `three`, `meta`) shows up in the error.Error() string?

Reply:
Also, possibly worth covering is the case where "two" and "three" have errors, but the "overall" error is nil? I'm not quite sure what the expectation would be in that case.

Reply:
Ah ha. Perfect. Let me play around a bit. Thank you for reviewing!